### PR TITLE
feat(curriculum-build): add meta schema and normalize meta

### PR DESCRIPTION
**Pull request by moT01**
This adds a schema validator for the meta.json files so there ~~isn't any inconsistencies~~ is less inconsistencies. 

It removes four things in the meta.json files:
`template` values that are an empty string
`required` values that are an empty array
`time` values that are an empty string
`superOrder` values

Quite sure they aren't needed.

Also, formats a few files to use two spaces instead of four.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Comment by moT01** [comment]
I'm gonna see if I can add some tests

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/get-challenges.js [hunk 1]
```diff
@@ -171,17 +172,17 @@ async function buildBlocks({ basename: blockName }, curriculum, superBlock) {
...
     const blockMeta = JSON.parse(fs.readFileSync(metaPath));
 
-    const { isUpcomingChange, helpCategory } = blockMeta;
-
-    if (typeof isUpcomingChange !== 'boolean') {
+    const validateMeta = metaSchemaValidator(blockMeta);
+    if (validateMeta.error) {
+      console.log(
+        `A validation error occurred in the meta.json for block '${blockName}'`
+      );
```
Comment (ShaunSHamilton) [inline curriculum/get-challenges.js:179]: I assume we want to remove this?

```suggestion
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
Has validation :heart:. A+ would review again.

**Review by ShaunSHamilton — APPROVED** [review]

**Comment by Muhammad-Mubashar516** [comment]
Hello, I am a new contributor and I would like to work on this issue. Can you please assign it to me? Thank you!

**Pull request merged**